src/kotlin/kage/Identity.kt

```diff
@@ -19,6 +20,21 @@ import kage.format.AgeStanza
  *
  * https://github.com/FiloSottile/age/blob/ab3707c085f2c1fdfd767a2ed718423e3925f4c4/age.go#L59-L72
  */
-public interface Identity {
-  public fun unwrap(stanzas: List<AgeStanza>): ByteArray
+public abstract class Identity {
+  internal abstract fun unwrapSingle(stanza: AgeStanza): ByteArray
+
+  public open fun unwrap(stanzas: List<AgeStanza>): ByteArray {
+    val lastError = IncorrectIdentityException()
+
+    stanzas.forEach { stanza ->
+      try {
```

Comment:
We are using exceptions for control flow a lot, for example `IncorrectIdentityException` is passed around and caught to try other identities. But I think this is a common idiom in java/kotlin...

Reply:
We've started using [kotlin-result](https://github.com/michaelbull/kotlin-result) in [APS](https://github.com/Android-Password-Store/Android-Password-Store) to model control flow in a more fluid way, you can experiment with it here as well.

Reply:
Yeah that looks good. I will take a look